### ISSUE TITLE
cmd/compile: Internal compiler error "missing wrapper for <type>"

### ISSUE THREAD

**Issue by misha-ridge** [golang/go#61040]
### What version of Go are you using (`go version`)?

<pre>
go version go1.21rc2 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes.

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=''
GOARCH='amd64'
GOBIN=''
GOCACHE='/home/misha/.cache/go-build'
GOENV='/home/misha/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMODCACHE='/home/misha/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/home/misha/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/home/misha/tectonic/_deps.linux-amd64/go-1.21rc2'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/home/misha/tectonic/_deps.linux-amd64/go-1.21rc2/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.21rc2'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='gcc'
CXX='g++'
CGO_ENABLED='1'
GOMOD='/home/misha/tmp/x/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build1654894307=/tmp/go-build -gno-record-gcc-switches'

</pre></details>

### What did you do?

```
$ go build ./...
./b/b.go:23:18: internal compiler error: missing wrapper for b.connContext
```

See https://github.com/misha-ridge/go-compiler-internal-error for the reproducer.

The error disappears if the files are moved into one package, so I can't provide go.dev/play link.

### What did you expect to see?

The code compiles.

### What did you see instead?

Internal compiler error

**Comment by ianlancetaylor** [comment]
This appears to be already fixed on HEAD.  I confirmed that it fails with 1.21rc2.

CC @randall77 @mdempsky @cuonglm just in case.

**Comment by misha-ridge** [comment]
Thanks. Will this be backported to 1.21 branch?

**Comment by cuonglm** [comment]
> Thanks. Will this be backported to 1.21 branch?

It will be included in 1.21 branch, as 1.21 is not released yet, so there's nothing to be backported.
